### PR TITLE
option, datapath: Move `AreDevicesRequired` to `option` package

### PR DESCRIPTION
**Pull request by pchaigno**
This will make it easier to reuse that helper function in other places, such as the loader in this pull request.

### CODE REVIEW

File: pkg/datapath/linux/devices.go [hunk 1]
```diff
@@ -403,16 +403,6 @@ func (dm *DeviceManager) Listen(ctx context.Context) (chan []string, error) {
...
-func (dm *DeviceManager) AreDevicesRequired() bool {
-	if option.Config.DryMode {
-		return false
-	}
-
-	return option.Config.EnableNodePort ||
-		option.Config.EnableHostFirewall ||
-		option.Config.EnableBandwidthManager
-}
-
```
Comment (julianwiedmann) [inline pkg/datapath/linux/devices.go:415]: Is it safe to drop the check for `option.Config.DryMode` here? If not we could just keep the function, but use `option.Config.AreDevicesRequired()` internally.

Comment (pchaigno) [reply pkg/datapath/linux/devices.go:415]: Yes, I don't think it really made sense to check that flag in the first place. I'll probably just send a PR to deprecate it.

**Review by julianwiedmann — APPROVED** [review]
ta!

**Review by joamaki — APPROVED** [review]

**Review by borkmann — APPROVED** [review]

**Review by pippolo84 — APPROVED** [review]
LGTM :100:

**Pull request merged**